azure-core/azure/core/pipeline/transport/requests_basic.py

```diff
@@ -117,29 +117,37 @@ def __init__(self, configuration=None, session=None):
         # type: (Optional[requests.Session]) -> None
         self._session_mapping = threading.local()
         self.config = configuration or Configuration()
-        self.session = session or requests.Session()
+        self.session = session
 
     def __enter__(self):
         # type: () -> RequestsTransport
-        self.config.connection.keep_alive = True
+        self.session = requests.Session()
```

Comment:
session value is already set in init?

Reply:
I've changed the Requests transport to match the aiohttp transport so that we only assign the session to the transport on entering a context manager.

Reply:
I see you have 
self.session = session
in init
and 
self.session = requests.Session()
in enter?

Reply:
Good call - it should be:
```
if not self.session:
    self.session = requests.Session()
```
And the same in aiohttp. I will fix and update.